tools/ci/actions/runner.sh

```diff
@@ -121,10 +121,23 @@ EOF
 
 }
 
+# ReportBuildStatus accepts an exit code as a parameter (defaults to 1) and also
+# instrucrts GitHub Actions to set build-status to 'failed' on non-zero exit or
+# 'success' otherwise.
+ReportBuildStatus () {
+  CODE=${1:-1}
```

Comment:
I'm unsure what this means, my best guess would be "variable 1, or `-1` if absent", why do we need a default here?

Reply:
You can't pass parameters to a `trap` function, so we have to work out a default.

`${parameter:-word}` is POSIX sh for a default value see [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02)). So this means "use the contents of `$1` unless its empty or unset in which case use `1`".

Reply:
An alternative would be to have a ReportBuildStatus with a proper parameter, and then parameter-less `ReportBuildSuccess` and `ReportBuildFailure` one-liners on top of it. But okay for the current code if we expect "real" shell programmers to be familiar with it.

Reply:
Who’re you accusing of being a “real” shell programmer 😉